clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -1427,6 +1447,14 @@ bool CheckAllArgsHaveFloatRepresentation(Sema *S, CallExpr *TheCall) {
                                   checkAllFloatTypes);
 }
 
+bool CheckArgIsFloatOrIntWithoutImplicits(Sema *S, Expr *Arg) {
+  auto checkFloat = [](clang::QualType PassedType) -> bool {
```

Comment:
LLVM style guide is this should be capitalized. Also wouldn't it make more sense as `CheckFloatOrInt`